app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -294,3 +315,351 @@ def test_hanging_protocol_clientside():
+            ),
+        ),
+    ),
+)
+@pytest.mark.parametrize(
+    "overlay_type",
+    (
+        InterfaceKindChoices.SEGMENTATION,
+        InterfaceKindChoices.STRING,
+    ),
+)
+@pytest.mark.parametrize(
+    "isolated_interface_type",
+    (
+        InterfaceKind.InterfaceKindChoices.CHART,
+        InterfaceKind.InterfaceKindChoices.PDF,
+    ),
+)
+@pytest.mark.parametrize(
+    "base_object_factory,base_object_lookup,object_factory,form_class,form_extra_kwargs",
+    (
+        (AlgorithmFactory, "algorithm", None, AlgorithmForm, {}),
+        (
+            PhaseFactory,
+            "phase",
+            None,
+            PhaseUpdateForm,
+            {"challenge": ChallengeFactory},
+        ),
+        (ArchiveFactory, "archive", ArchiveItemFactory, ArchiveForm, {}),
+        (
+            ReaderStudyFactory,
+            "reader_study",
+            DisplaySetFactory,
+            ReaderStudyUpdateForm,
+            {},
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_forms_view_content_help_text(
+    number_of_images,
+    number_of_overlays,
+    number_of_isolated_interfaces,
+    number_of_undisplayable_interfaces,
+    overlay_type,
+    isolated_interface_type,
+    expected_help_text,
+    base_object_factory,
+    base_object_lookup,
+    object_factory,
+    form_class,
+    form_extra_kwargs,
+):
+    ci_list = []
+    civ_list = []
+
+    for i in range(number_of_isolated_interfaces):
+        ci = ComponentInterfaceFactory(
+            kind=isolated_interface_type,
+            title=f"test-ci-isolated-{i}",
+        )
+        civ = ComponentInterfaceValueFactory(
+            interface=ci,
+        )
+        ci_list.append(ci)
+        civ_list.append(civ)
+
+    for i in range(number_of_images):
+        ci = ComponentInterfaceFactory(
+            kind=InterfaceKindChoices.IMAGE,
+            title=f"test-ci-image-{i}",
+        )
+        civ = ComponentInterfaceValueFactory(
+            interface=ci,
+        )
+        ci_list.append(ci)
+        civ_list.append(civ)
+
+    for i in range(number_of_overlays):
+        ci = ComponentInterfaceFactory(
+            kind=overlay_type,
+            title=f"test-ci-overlay-{i}",
+        )
+        civ = ComponentInterfaceValueFactory(
+            interface=ci,
+        )
+        ci_list.append(ci)
+        civ_list.append(civ)
+
+    for i in range(number_of_undisplayable_interfaces):
+        ci = ComponentInterfaceFactory(
+            kind=InterfaceKind.InterfaceKindChoices.ZIP,
+            title=f"test-ci-undisplayable-{i}",
+        )
+        civ = ComponentInterfaceValueFactory(
+            interface=ci,
+        )
+        ci_list.append(ci)
+        civ_list.append(civ)
+
+    base_obj = base_object_factory()
+
+    if object_factory:
+        object = object_factory(
+            **{base_object_lookup: base_obj},
```

Comment:
The base_object doesn't matter for this test, right? You can just create an ArchiveItem or DisplaySet and then reference the automatically created base object in the form with `object.base_object`. You also don't need to explicitly create a challenge for the Phase, that is done by the factory for you.

Reply:
For the first point about `base_object` that is true, but then we need to move the form and base_obj to inside the if statement (and duplicate the form part) to deal with each case separately. I was trying to minimize the difference as much as possible. So, currently it is:

```python
    base_obj = base_object_factory()

    if object_factory:
        object = object_factory(
            **{base_object_lookup: base_obj},
        )
        object.values.set(civ_list)
    else:
        base_obj.inputs.set(ci_list)
        base_obj.outputs.set([])

    form = form_class(
        user=UserFactory(), instance=base_obj, **form_extra_kwargs
    )
```

After the suggestion, I have this in mind now:

```python
    if object_factory:
        object = object_factory()
        object.values.set(civ_list)

        form = form_class(
            user=UserFactory(), instance=object.base_object,
        )
    else:
        base_obj = base_object_factory()

        base_obj.inputs.set(ci_list)
        base_obj.outputs.set([])

        form = form_class(
            user=UserFactory(), instance=base_obj, **form_extra_kwargs
        )
```

 Am I missing something? and which one you prefer?

Reply:
Also realized, since we need to use `algorithm_inputs` to set the ci_list to the `Phase` (based on another comment you made), then the property name `inputs`/`algorithm_inputs` should also be passed as a parameter to the test.

Would it be better to move the Phase case to its own test?

Reply:
Ok, given the complexity, I think it would be better to create separate tests after all. 
- One test for Archives and ReaderStudies (you can parametrize that test without the need for if statements)
- One test for Algorithms
- One test for Phases

It will introduce 3 very similar tests, but they will each be easier to read.

Reply:
Thanks! I agree.
Done